addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -280,8 +280,21 @@ def _compute_partner_phone_email(self):
                 applicant.partner_mobile = applicant.partner_id.mobile
 
     def _inverse_partner_email(self):
-        for applicant in self.filtered(lambda a: a.partner_id and a.email_from):
-            applicant.partner_id.email = applicant.email_from
+        for applicant in self.filtered(lambda a: a.email_from):
+            if not applicant.partner_id:
+                if not (applicant.partner_name or applicant.name):
```

Comment:
`applicant.name`, for some reason, is the job position name. We shouldn't use it here and rely only on the partner_name field.
```suggestion
                if not applicant.partner_name:
```